File: examples/typescript/solana/funding/account.fund.ts
// Usage: pnpm tsx solana/funding/account.fund.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";
async function main() {
  const cdp = new CdpClient();
  const account = await cdp.solana.getOrCreateAccount({ name: "account" });

  const fundOperation = await account.fund({
    token: "sol",
    amount: 10000000n, // 0.01 sol
  });

  const completedTransfer = await account.waitForFundOperationReceipt({
    transferId: fundOperation.id,
  });

  console.log(completedTransfer);
}

main().catch(console.error);

File: examples/python/solana/funding/account.fund.py
# Usage: uv run python solana/funding/account.fund.py

import asyncio
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        account = await cdp.solana.get_or_create_account(name="account")

        fund_operation = await account.fund(
            token="sol",
            amount=10000000, # 0.01 sol
        )

        completed_transfer = await account.wait_for_fund_operation_receipt(
            transfer_id=fund_operation.id,
        )

        print(completed_transfer)


asyncio.run(main())
